### PR TITLE
Make our vim java indents more consistent with IntelliJ

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Make our vim java indents more consistent with IntelliJ by:
- `tabstop=2 shiftwidth=2 softtabstop=2`: tabs are 2 spaces
- `cinoptions` `+1s`: continuation indents are 1x shiftwidth (i.e. 2 spaces)
- `cinoptions` `(0`: continuation indents in function argument lists align on new lines
- `cinoptions` `W1s`: continuation indents in function argument where the
first item is on the next line fallback to standard continuation
indenting

### Why
Writing java in our vim config is significantly different to our
IntelliJ config. These changes make some basic improvements, more are
needed, but this makes vim do the right thing in many more cases than
it does now.

### Checklist
- [ ] ~Send RFC email to Braintree developers _if change may be impactful_. Please include a link to this PR so that discussion about the pros and cons of the change remains linked to the proposed changes. **Recent examples include**: addition of linter and completer, no longer removing end-of-line whitespace on save, changing to fzf for file lookup.~

### CODE REVIEW

File: vimrc [hunk 1]
```diff
@@ -79,7 +79,7 @@ augroup END
 
 autocmd FileType php setlocal tabstop=4 shiftwidth=4 softtabstop=4
 autocmd FileType python setlocal tabstop=4 shiftwidth=4 softtabstop=4
-autocmd FileType java setlocal tabstop=4 shiftwidth=4 softtabstop=4
+autocmd FileType java setlocal expandtab tabstop=2 shiftwidth=2 softtabstop=2
```
Comment (pgr0ss) [inline vimrc]: This is the same as our default, right? Does this mean you delete this line entirely?

https://github.com/braintreeps/vim_dotfiles/blob/9826007dbdfac70728e55e4f0d76fe7b43754e25/vimrc#L38

Comment (leighmcculloch) [reply vimrc]: Ah good catch. I think you're right. I'll just send the line.

Comment (leighmcculloch) [reply vimrc]: 6a5b6ff

**Comment by leighmcculloch** [comment]
Actually, this isn't going to work as is because the Java SDK uses a different set of indent and continuation indent behavior. I doubt we'd adopt per project vimrc's because there are security concerns with doing so, so maybe we can have these config values activated only if an `.ijwd` folder is present in the current directory which could be the signal that IntelliJ is being used for the project and therefore our IntelliJ indents should apply?

**Comment by leighmcculloch** [comment]
I have updated the PR so that the indent changes that would be incompatible with the projects we don't use IntelliJ don't apply. I reintroduced explicitly stating Java is 2 characters so the fork between the two is clearer.

**Comment by leighmcculloch** [comment]
Actually, it looks like we have some Java repos that are using two spaces and don't have a `.ijwb` directory...

**Comment by leighmcculloch** [comment]
Okay, I checked a lot of repos and the only one I could find using 4 characters is the Java SDK, so I think we can just fork on that and assume 2 characters for everything else. If one person could give a final 👀 and 👍 I'll merge it.

**Pull request merged**